### PR TITLE
Use ts strict checking for binding modbus

### PR DESCRIPTION
**Pull request by relu91**
Yet another down. 

Relates to #758.

### CODE REVIEW

File: packages/binding-modbus/src/modbus-connection.ts [hunk 1]
```diff
@@ -461,12 +483,21 @@ export class PropertyOperation {
     done(base?: number, buffer?: Buffer): void {
         debug("Operation done");
 
+        if (!this.resolve || !this.reject) {
+            throw new Error("Invoked done before executing operation");
```
Comment (danielpeintner) [inline packages/binding-modbus/src/modbus-connection.ts:487]:
```suggestion
            throw new Error("Invoke done before executing operation");
```

I think that's what you meant, right?
OR

`Invoke called before executing operation`

File: packages/binding-modbus/src/modbus-connection.ts [hunk 2]
```diff
@@ -490,6 +521,9 @@ export class PropertyOperation {
      */
     failed(reason: Error): void {
         warn(`Operation failed: ${reason}`);
+        if (!this.reject) {
+            throw new Error("Invoked reject before executing operation");
```
Comment (danielpeintner) [inline packages/binding-modbus/src/modbus-connection.ts:525]: Again not sure about the _passive_ form of invoked...

**Review by danielpeintner — COMMENTED** [review]
Looks good in general.

Have some understanding problems w.r.t. some error messages. See below

File: packages/binding-modbus/src/modbus-connection.ts (hunk 1 above)
Comment (relu91) [reply packages/binding-modbus/src/modbus-connection.ts:487]: Mmm not sure, what I meant is that the function `done` was invoked before executing the operation. Maybe it is better to be more explicit here: `Invoked function "done" before executing the Modbus operation` ?

Comment (danielpeintner) [reply packages/binding-modbus/src/modbus-connection.ts:487]: Your proposal makes sense.
An alternative *could* be 

`Function "done" was executed before the (this?) Modbus operation`

Either way is fine by me 👍

File: packages/binding-modbus/src/modbus-connection.ts (hunk 2 above)
Comment (danielpeintner) [reply packages/binding-modbus/src/modbus-connection.ts:525]: see comment above

**Comment by danielpeintner** [comment]
> ❗ Your organization is not using the GitHub App Integration. As a result you may experience degraded service beginning May 15th. Please [install the Github App Integration](https://github.com/apps/codecov) for your organization. [Read more](https://about.codecov.io/blog/codecov-is-updating-its-github-integration/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

I tried already several times to activate the GitHub integration but I do not know "what" blocks it?

image

Shall I open a ticket on Eclipse side?

**Comment by danielpeintner** [comment]
> Shall I open a ticket on Eclipse side?

FYI: I opened https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/3545

File: packages/binding-modbus/src/modbus-connection.ts (hunk 1 above)
Comment (relu91) [reply packages/binding-modbus/src/modbus-connection.ts]: I changed it a little bit more in https://github.com/eclipse-thingweb/node-wot/pull/1050/commits/dbf99795f515e01401a5d17ab1a460046cb2f0e3. I hope it is ok, merge if it is ok! 🛩️

**Comment by relu91** [comment]
@danielpeintner do you think we can merge it even if we have redcrosses in codecov?

**Review by danielpeintner — APPROVED** [review]

**Comment by danielpeintner** [comment]
> @danielpeintner do you think we can merge it even if we have redcrosses in codecov?

We are now seeing these kind of "errors" in all our PRs now. I think we need to properly configure it and maybe show it as a warning or so.

Anyhow, w.r.t. this PR I think we can merge as is

**Pull request merged**